risc0/groth16/src/prove/cuda.rs

```diff
@@ -193,7 +193,7 @@ struct CalcWitness {
 impl CalcWitness {
     fn as_bytes(&self) -> &[u8] {
         let r: &[wtns_file::FieldElement<32>] = &self.witness[..];
-        unsafe { std::slice::from_raw_parts(r.as_ptr() as *const u8, r.len() * 4) }
+        unsafe { std::slice::from_raw_parts(r.as_ptr() as *const u8, std::mem::size_of_val(r)) }
```

Comment:
`size_of_val` appears to only return the size of a single element
```suggestion
        unsafe { std::slice::from_raw_parts(r.as_ptr() as *const u8, std::mem::size_of_val(r) * r.len()) }
```

https://doc.rust-lang.org/std/mem/fn.size_of_val.html

Reply:
I think you are mistaken, look at what it says about `[T]`.

also here is a playground for proof: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=a90fae17ac7ed0dfbbb66b2b7b4c3197